### PR TITLE
chore: update readme with sql admin and client role details

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Adding extra details to README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,18 +35,30 @@ your Cloud SQL instance.
...
 To explicitly set a specific source for the Credentials to use, see [Using
 Option](#using-options) below.
 
+The source identity of the credentials (user, service account, etc.) must have
+the [Cloud SQL Client][client-role] IAM Role in order to successfully 
+make Cloud SQL connections.
+
 [adc]: https://cloud.google.com/docs/authentication
 [google-auth]: https://pkg.go.dev/golang.org/x/oauth2/google#hdr-Credentials
+[client-role]: https://console.cloud.google.com/iam-admin/roles/details/roles%3Ccloudsql.client
```
Comment (enocom) [inline README.md:61]: This is probably the link to use: https://cloud.google.com/sql/docs/mysql/roles-and-permissions.

**Review by enocom — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -35,18 +35,30 @@ your Cloud SQL instance.
...
 [golang.org/x/oauth2/google][google-auth] documentation for more information in
 how these credentials are sourced.
 
 To explicitly set a specific source for the Credentials to use, see [Using
 Option](#using-options) below.
 
+The source identity of the credentials (user, service account, etc.) must have
+the [Cloud SQL Client][client-role] IAM Role in order to successfully 
+make Cloud SQL connections.
+
```
Comment (kurtisvg) [inline README.md:58]: Can we figure out how to work this higher? Essentially we want to lead with "You must have a IAM member with this role" then say "here's how we figure out which IAM member we use".

File: README.md [hunk 3]
```diff
@@ -35,15 +35,30 @@ your Cloud SQL instance.
...
+the IAM Principal.
+
+[admin-api]: https://console.cloud.google.com/apis/api/sqladmin.googleapis.com
+[client-role]: https://cloud.google.com/sql/docs/mysql/roles-and-permissions
+
 ### Credentials
 
 This repo uses the [Application Default Credentials (ADC)][adc] strategy for
-typing providing credentials. Please see the
+resolving credentials from an IAM Principal. Please see the
```
Comment (enocom) [inline README.md:55]:
```suggestion
resolving credentials. Please see the
```

File: README.md [hunk 4]
```diff
@@ -35,15 +35,30 @@ your Cloud SQL instance.
 The instance connection name for your Cloud SQL instance is always in the
 format "project:region:instance".
 
+### APIs and Services
+
+This package requires the following to successfully make Cloud SQL Connections:
+
+- IAM Principal (user, service account, etc.) with the
+[Cloud SQL Client][client-role] role. This IAM Principal will be used for
```
Comment (kurtisvg) [inline README.md:43]: This should be "[Cloud SQL Client][client-role] role or equivalent".

**Review by kurtisvg — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -35,15 +35,30 @@ your Cloud SQL instance.
 The instance connection name for your Cloud SQL instance is always in the
 format "project:region:instance".
 
+### APIs and Services
+
+This package requires the following to successfully make Cloud SQL Connections:
+
+- IAM Principal (user, service account, etc.) with the
```
[the remark this answers is not in the document]
Comment (enocom) [reply README.md:42]: That seems to be how it's used here: https://cloud.google.com/iam/docs/overview.

**Pull request merged**